tests/e2e/api/controllers/_changes.spec.js

```diff
@@ -741,77 +661,72 @@ describe('changes handler', () => {
-                      .request({
-                        path: `/_users/org.couchdb.user:steve?rev=${user._rev}`,
-                        method: 'PUT',
-                        json: true,
-                        body: _.extend(user, { facility_id: 'fixture:bobville' })
-                      })
-                      .then(() => Promise.all([
-                        utils.saveDoc(_.extend(medicUser, { facility_id: 'fixture:somethingville', roles: ['_admin'] })),
-                        utils.saveDocs(allowedBob),
-                        utils.saveDocs(allowedSteve),
-                      ]))
-                      .then(resolve);
-                  }, 1000);
-                })
-              ]);
-            })
-            .then(([ changes ]) => {
-              if (changes.results.find(change => change.id === 'org.couchdb.user:steve')) {
-                expect(changes.results.every(change => bobsIds.indexOf(change.id) !== -1 ||
-                                                       change.id === 'org.couchdb.user:steve' ||
-                                                       change.id === 'settings')
-                ).toBe(true);
-              }
-            })
-            .then(() => utils.request({ path: '/_users/org.couchdb.user:steve' }))
-            .then(user => utils.request({
-              path: `/_users/org.couchdb.user:steve?rev=${user._rev}`,
-              method: 'PUT',
-              json: true,
-              body: _.extend(user, { facility_id: 'fixture:steveville' })
-            }));
-        });
+      const updateSteve = (user, medicUser) => {
+        // we move Steve's _user to bobville
+        return utils
+          .request({
+            path: `/_users/org.couchdb.user:steve?rev=${user._rev}`,
+            method: 'PUT',
+            body: _.extend(user, { facility_id: 'fixture:bobville' })
+          })
+          .then(() => utils.saveDocs([
+            // we "move" his medic user to a place that doesn't exist. He should still get bobville docs
+            _.extend(medicUser, { facility_id: 'fixture:somethingville', roles: ['_admin'] }),
+            ...allowedSteve,
+            ...allowedBob,
+          ]));
+      };
+
+      return getCurrentSeq('steve')
+        .then(() => Promise.all([
+          utils.request({ path: '/_users/org.couchdb.user:steve' }),
+          utils.getDoc('org.couchdb.user:steve')
+        ]))
+        .then(([ user, medicUser ]) => Promise.all([
+          requestChanges('steve', { feed: 'longpoll', since: currentSeq }),
+          utils.delayPromise(() => updateSteve(user, medicUser), 500),
+        ]))
+        .then(([ changes ]) => {
+          const bobvilleIds = getIds(allowedBob);
+          const stevevilleIds = getIds(allowedSteve);
+          const changeIds = getIds(changes.results);
+
+          // Sometimes the changes request ends before all our changes are processed and doesn't contain the
+          // update to Steve's user :(
```

Comment:
I kept thinking of ways to make the test more stable but I really can't control how changes are received by the controller :(